### PR TITLE
README.md: Document CMake build support

### PR DESCRIPTION
**Pull request by LDong-Arm**
Preceding PR: #26 

(The last commit belongs to this PR.)
Document how to build greentea-client in another CMake project and set up a stream of I/O.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,3 +73,48 @@ You can find the greentea-client example application [here](./example/main.cpp).
     cmake --build cmake_build
 
 This generates an executable `./cmake_build/example/greentea-client-example`.
+
+# Adding greentea-client to a project
```
Comment (LDong-Arm) [inline README.md:67]: @Patater @rwalton-arm Please review whether this section is consistent with how we integrate greentea-client with mbed-os, thanks.

File: README.md [hunk 2]
```diff
@@ -68,3 +73,48 @@ You can find the greentea-client example application [here](./example/main.cpp).
...
+
+## Build support
+
+greentea-client supports the CMake build system. To add greentea-client to a project, call
+[`add_subdirectory`](https://cmake.org/cmake/help/latest/command/add_subdirectory.html)
+pointing to the path to the client. Then link `greentea-client` to a CMake target that requires it with
```
Comment (Patater) [inline README.md]:
```diff
- pointing to the path to the client.
+ pointing to the relative path of the `greentea-client` project directory.
```

File: README.md [hunk 3]
```diff
@@ -68,3 +73,48 @@ You can find the greentea-client example application [here](./example/main.cpp).
...
+[`target_link_libraries`](https://cmake.org/cmake/help/latest/command/target_link_libraries.html).
```
Comment (Patater) [inline README.md:75]: Having examples of both `add_subdirectory(greentea-client)` and `target_link_libraries(my_app greentea-client)` would help with clarity.

File: README.md [hunk 4]
```diff
@@ -68,3 +73,48 @@ You can find the greentea-client example application [here](./example/main.cpp).
...
+
+# Add greentea-client
+add_subdirectory(greentea-client)
+
+# Add custom I/O implementation
+target_sources(greentea-client
+    PRIVATE
+        custom_test_io.cpp
+)
+```
```
Comment (Patater) [inline README.md]: Could we make a custom IO example (our second example) and link to it instead of adding untested, may-go-stale code in the README?

**Review by Patater — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (rwalton-arm) [reply README.md:67]: One small issue that I can see: CMake isn't a "build system", it's a build system generator. The "build system" would be Ninja, make etc.

Comment (LDong-Arm) [reply README.md:67]: Done. I changed it to simply "supports CMake".

File: README.md (hunk 2 above)
Comment (LDong-Arm) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (LDong-Arm) [reply README.md:75]: Added

File: README.md (hunk 4 above)
Comment (LDong-Arm) [reply README.md]: Added

**Comment by LDong-Arm** [comment]
@Patater @rwalton-arm Thanks for the suggestions, I've added an example to demonstrate a custom IO based on regular files, please review.

File: CMakeLists.txt [hunk 5]
```diff
@@ -32,10 +32,14 @@ if(GREENTEA_CLIENT_STDIO)
...
 
 set(BUILD_EXAMPLES ON CACHE BOOL "Enable building the examples")
 
 if((CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME) AND BUILD_EXAMPLES)
-    add_subdirectory(example)
+    if(GREENTEA_CLIENT_STDIO)
+        add_subdirectory(examples/stdio)
+    else()
+        add_subdirectory(examples/custom_io)
+    endif()
```
Comment (LDong-Arm) [inline CMakeLists.txt:44]: CMake question: Is it possible to build two applications at once, with two different configurations (GREENTEA_CLIENT_STDIO ON/OFF) of a static libraries? I know it's doable if both applications use same configuration of the library...

Comment (Patater) [reply CMakeLists.txt:44]: I'm not aware of a way to do that elegantly. Inelegantly, you could write a Makefile which calls `cmake` with each set of configuration options and builds the targets you want. Another approach is to specify two different `build` directories, one for each configuration set, and then e.g. `cmake --build build_stdio` and `cmake --build build_nostdio`.

File: examples/stdio/main.c [hunk 6]
```diff
@@ -14,12 +14,13 @@
...
  */
 
+#include <stdio.h>
 #include "greentea-client/test_env.h"
 
 int main()
 {
     const char *key = "demo_key";
-    const int value = 60;
+    const char *value = "60";
```
Comment (Patater) [inline examples/stdio/main.c]: Why the change to a string? Shall we explain in a standalone commit's message?

File: CMakeLists.txt (hunk 5 above)
Comment (LDong-Arm) [reply CMakeLists.txt:44]: Fair enough. I guess building two versions of the same target in one go is not a supported/intended use case of CMake. I'll keep this unchanged.

File: examples/stdio/main.c (hunk 6 above)
Comment (LDong-Arm) [reply examples/stdio/main.c]: I converted this example to C, so overloads of `greentea_send_kv` are unavailable and the C version of it takes a string.

Maybe I should convert it back to C++, to allow demonstration of the full API?

Comment (LDong-Arm) [reply examples/stdio/main.c]: Done, I've converted the files back to C++

Comment (Patater) [reply examples/stdio/main.c]: I didn't pick up that C only allowed strings. I am okay with or without this change, just we need a commit message to explain it if we want to make the change.

Comment (LDong-Arm) [reply examples/stdio/main.c]: I've already converted it back to C++ because it allows us to use the full API. Maybe in the future we can send several KV pairs of different types (string, int, multiple ints for reporting passes & failures, ...) in our examples to demonstrate this. So does the current state of the PR look okay?

Comment (Patater) [reply examples/stdio/main.c]: Yup, all fine.

**Review by Patater — APPROVED** [review]

**Review by rwalton-arm — APPROVED** [review]
